types/queries/update_word.sql

```diff
@@ -1 +1,13 @@
-update word set source_text = $2 where id = $1
+update word set
+    source_text =
+        case
+            when $2::text[] != '{}' and $2[1] is not null then $2[1]
+            else source_text
+        end,
+    commentary =
+        case
+            when $3::text[][1] is not null then $3[1]
+            when $3[1] is null then null
```

Comment:
I think this can be simplified down to:

```sql
when $3::text[] != '{}' then $3[1]
```

since we just want whatever the value is if the list isn't empty.